### PR TITLE
feat(aws-lambda): implement new designs

### PR DESCRIPTION
**Pull request by scefali**
This PR implements some of the new designs we got from John for the AWS Lambda integration. I had to modify some of the form logic so we could make things look more like the designs. Also added some tests that were missing.

Before:
Screen Shot 2021-01-20 at 12 47 07 PM

After:
Screen Shot 2021-01-20 at 3 49 34 PM

Before:
Screen Shot 2021-01-20 at 12 49 24 PM

After:
Screen Shot 2021-01-20 at 12 55 54 PM

Before:
Screen Shot 2021-01-20 at 12 51 22 PM

After:
Screen Shot 2021-01-20 at 12 58 47 PM

### CODE REVIEW

**Review by NisanthanNanthakumar — APPROVED** [review]

**Comment by EvanPurkhiser** [comment]
Can we get rid of this bottom border?

**Comment by EvanPurkhiser** [comment]
image

Spacing here around this button is a bit weird

File: src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx [hunk 1]
```diff
@@ -114,28 +129,47 @@ const Item = styled('div')`
...
+  padding-bottom: ${space(1)};
+`;
+
+const RuntimeAndVersion = styled('div')`
+  display: flex;
+  flex-direction: row;
+  color: ${p => p.theme.gray300};
+`;
+
+//match the line eight of the badge
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:172]:
```suggestion
//match the line height of the badge
```

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaProjectSelect.tsx [hunk 2]
```diff
@@ -22,29 +40,29 @@ export default class AwsLambdaProjectSelect extends React.Component<Props> {
...
     // TODO: Add logic if no projects
     return (
-      <StyledForm model={this.model} onSubmit={handleSubmit}>
-        <JsonForm forms={[formFields]} />
-      </StyledForm>
+      <Wrapper>
+        <StyledForm model={this.model} hideFooter>
+          <JsonForm forms={[formFields]} />
+        </StyledForm>
+        <Observer>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaProjectSelect.tsx:55]: mobx expert

File: src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+type Props = {buttonText: string; docsUrl: string} & Pick<
+  React.ComponentProps<typeof Button>,
+  'disabled' | 'onClick'
+>;
+
+export default function FooterWithButtons({buttonText, docsUrl, ...rest}: Props) {
+  return (
+    <Footer>
+      <ButtonWrapper>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx:16]: Can you use `ButtonBar` here and get rid of `StyledButton`

File: src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+`;
+
+const StyledButton = styled(Button)`
+  padding: 0;
+  margin-right: ${space(2)};
+`;
+
+const ButtonWrapper = styled('div')`
+  padding: ${space(2)} 0;
+  float: right;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx:45]: maybe kill the float and set `Footer` to `display: flex` and `align-items: flex-end`

File: src/sentry/static/sentry/app/views/integrationPipeline/components/iconGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import styled from '@emotion/styled';
+
+import {IconSentry} from 'app/icons';
+import PluginIcon from 'app/plugins/components/pluginIcon';
+
+export default function IconGroup({pluginId}: {pluginId: string}) {
+  return (
+    <IconWrapper>
+      <StyledPluginIcon size={50} pluginId={pluginId} />
+      <StyledIconSentry />
```
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/static/sentry/app/views/integrationPipeline/components/iconGroup.tsx:11]: @EvanPurkhiser unfortunately, `size` can only be the following:
```
const iconSizes = {
  xs: '12px',
  sm: '16px',
  md: '20px',
  lg: '24px',
  xl: '32px',
  xxl: '72px',
};
```

[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/integrationPipeline/components/iconGroup.tsx:11]: damn. Do you have any qualms of just swapping these down to 32px lol

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaFunctionSelect.tsx [hunk 6]
```diff
@@ -62,3 +103,17 @@ const StyledForm = styled(Form)`
   margin: 50px;
   padding-bottom: 50px;
 `;
+
+const Wrapper = styled('div')``;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaFunctionSelect.tsx:107]: can you delete this now?

File: src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaProjectSelect.tsx [hunk 7]
```diff
@@ -53,3 +72,5 @@ const StyledForm = styled(Form)`
   max-width: 500px;
   margin: 50px;
 `;
+
+const Wrapper = styled('div')``;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaProjectSelect.tsx:76]: Should it even be in a `div`? Maybe just `React.Fragment`?

File: src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  display: flex;
+  justify-content: flex-end;
+  bottom: 0;
+  z-index: 100;
+  background-color: ${p => p.theme.bodyBackground};
+  border-top: 1px solid ${p => p.theme.gray100};
+`;
+
+const StyledButtonBar = styled(ButtonBar)`
+  padding: ${space(2)};
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx:42]: Why not put the padding on the footer? and then you don't need this styled component right

File: src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+`;
+
+const StyledButton = styled(Button)`
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/integrationPipeline/components/footerWithButtons.tsx:45]: Wait would it just make sense to use `size="xsmall"`?

File: src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx [hunk 10]
```diff
@@ -83,20 +83,35 @@ class IntegrationServerlessRow extends React.Component<Props> {
...
+  renderLayerStatus() {
+    const {serverlessFunction} = this.props;
+    if (serverlessFunction.outOfDate) {
+      return (
+        <UpdateButton size="small" priority="primary" onClick={this.updateVersion}>
+          {t('Update')}
+        </UpdateButton>
+      );
+    }
+    return this.enabled ? t('Latest') : t('Disabled');
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:95]: total nit: But I would swap these conditionals, minorly easier to read when the if condition is just a one line return

**Review by evanpurkhiser — APPROVED** [review]

File: src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx (hunk 10 above)
Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:95]: eh, I think this is easier to read in multiple lines

File: src/sentry/static/sentry/app/views/integrationPipeline/components/iconGroup.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/static/sentry/app/views/integrationPipeline/components/iconGroup.tsx]: nah, I'll do that

File: src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx (hunk 10 above)
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:95]:
```jsx
  renderLayerStatus() {
    const {serverlessFunction} = this.props;
    if (!serverlessFunction.outOfDate) {
		return this.enabled ? t('Latest') : t('Disabled');
	}

    return (
      <UpdateButton size="small" priority="primary" onClick={this.updateVersion}>
        {t('Update')}
      </UpdateButton>
    );
}
```

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:95]: Is what I meant, less pyramid-y

Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:95]: oh got it, yea

Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:95]: Like I said tho, totally a nit lol

Comment (scefali) [reply src/sentry/static/sentry/app/views/organizationIntegrations/integrationServerlessRow.tsx:95]: I have a test failing so I change this as well :)

**Pull request merged**